otcextensions/sdk/smn/v2/_proxy.py

```diff
@@ -9,9 +9,306 @@
+        """Delete all attributes of a topic
+
+        :param topic: Either the ID of a topic or a
+            :class:`~otcextensions.sdk.smn.v2.topic.Topic`
+            instance.
+        :param name: Attribute Name.
+        :returns: request_id.
+        :rtype: :class:`~otcextensions.sdk.smn.v2.topic.TopicAttribute`
+        :raises: :class:`~openstack.exceptions.ResourceNotFound`
+            when no resource can be found.
+        """
+        topic = self._get_resource(_topic.Topic, topic)
+        if name:
+            return self._delete(_topic.TopicAttribute,
+                                id=name, topic_id=topic.id)
+        return self._delete(_topic.TopicAttribute,
+                            topic_id=topic.id, requires_id=False)
+
+    # ======== Subscription ========
+    def create_subscription(self, topic, **attrs):
+        """Create a new Subscription from attributes
+
+        :param topic: Either the ID of a topic or a
+            :class:`~otcextensions.sdk.smn.v2.topic.Topic`
+            instance.
+        :param dict attrs: Keyword arguments which will be used to create
+            a :class:`~otcextensions.sdk.smn.v2.subscription.Subscription`
+        :returns: :class:`~otcextensions.sdk.smn.v2.subscription.Subscription`
+        """
+        topic = self._get_resource(_topic.Topic, topic)
+        return self._create(_subscription.Subscription,
+                            topic_urn=topic.id, **attrs)
+
+    def delete_subscription(self, subscription, ignore_missing=True):
+        """Delete a subscription
+
+        :param subscription: Either the ID of a subscription or a
+            :class: `~otcextensions.sdk.smn.v2.subscription.Subscription`
+            instance.
+        :param bool ignore_missing: When set to ``False``
+            :class:`~openstack.exceptions.ResourceNotFound` will be raised
+            if the subscription does not exist.
+            When set to ``True``, no exception will be set when attempting to
+            delete a nonexistent subscription.
+        :returns: ``None``
+        """
+        return self._delete(_subscription.Subscription, subscription,
+                            ignore_missing=ignore_missing)
+
+    def subscriptions(self, topic=None, **query):
+        """Return a generator of Subscriptions
+
+        :param topic: Either the ID of a topic or a
+            :class: `~otcextensions.sdk.smn.v2.topic.Topic`
+            instance.
+        :param dict query: Optional query parameters to be sent to limit
+            the resources being returned.
+        :returns: A generator of subscription objects
+        :rtype: :class:`~otcextensions.sdk.smn.v2.subscription.Subscription`
+        """
+        if topic:
+            topic = self._get_resource(_topic.Topic, topic)
+            return self._list(_subscription.Subscription,
+                              topic_urn=topic.id, **query)
+        base_path = '/notifications/subscriptions'
```

Comment:
you can override list for subscriptions, for not to store this hardcoded parameter in proxy

Reply:
But how can I solve the "if" cause then?

Reply:
in self._list check topic_urn, btw you can skip this comment, this is just idea)

Reply:
ok, done